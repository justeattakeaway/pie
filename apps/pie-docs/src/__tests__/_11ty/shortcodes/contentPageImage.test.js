const contentPageImage = require('../../../_11ty/shortcodes/contentPageImage');

describe('contentPageImage.js', () => {
    it.each([
        {
            width: '200px',
            alt: 'foo bar',
            src: 'some/path/to/image',
            mobileSrc: 'some/path/to/mobile/image',
            context: 'someContainer',
            caption: 'some caption text [with a link](fakeurl/path)'
        },
        {
            src: 'some/path/to/image'
        }
    ])('creates the expected markup for the given config settings', config => {
        const result = contentPageImage(config);

        expect(result).toMatchSnapshot();
    });
});
