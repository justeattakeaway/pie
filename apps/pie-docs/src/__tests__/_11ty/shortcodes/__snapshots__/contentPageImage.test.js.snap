// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contentPageImage.js > creates the expected markup for the given config settings 1`] = `
"<figure class=\\"c-someContainer-img c-contentImage c-contentImage--hasBackdrop\\">
        <div class=\\"c-contentImage-backdrop c-contentImage-default\\">
          <picture>
            <source style=\\"--img-width: 200px;\\" media=\\"(max-width: 600px)\\" srcset=\\"some/path/to/mobile/image\\">
            <img src=\\"some/path/to/image\\" style=\\"--img-width: 200px;\\" alt=\\"foo bar\\">
          </picture>
        </div>
        <figcaption class=\\"c-contentImage-caption\\"><span>some caption text <a href=\\"fakeurl/path\\">with a link</a></span></figcaption>
      </figure>"
`;

exports[`contentPageImage.js > creates the expected markup for the given config settings 2`] = `
"<figure class=\\"c-contentPage-img c-contentImage c-contentImage--fullWidth\\">
        <div class=\\"c-contentImage-backdrop c-contentImage-default\\">
          <picture>
            
            <img src=\\"some/path/to/image\\"  alt=\\"\\">
          </picture>
        </div>
        
      </figure>"
`;
